backends/tfhe-cuda-backend/cuda/src/polynomial/functions.cuh

```diff
@@ -162,28 +163,17 @@ __device__ void round_to_closest_multiple_inplace(T *rotated_acc, int base_log,
 template <typename Torus, class params>
 __device__ void add_to_torus(double2 *m_values, Torus *result,
                              bool init_torus = false) {
-  Torus mx = (sizeof(Torus) == 4) ? UINT32_MAX : UINT64_MAX;
   int tid = threadIdx.x;
 #pragma unroll
   for (int i = 0; i < params::opt / 2; i++) {
     double v1 = m_values[tid].x;
     double v2 = m_values[tid].y;
 
-    double frac = v1 - floor(v1);
-    frac *= mx;
-    double carry = frac - floor(frac);
-    frac += (carry >= 0.5);
-
     Torus V1 = 0;
-    typecast_double_to_torus<Torus>(frac, V1);
-
-    frac = v2 - floor(v2);
-    frac *= mx;
-    carry = frac - floor(v2);
-    frac += (carry >= 0.5);
+    typecast_double_round_to_torus<Torus>(v1, V1);
 
     Torus V2 = 0;
-    typecast_double_to_torus<Torus>(frac, V2);
+    typecast_double_round_to_torus<Torus>(v2, V2);
```

Comment:
may I suggest more readable variable names ?

Reply:
Do you have something in mind? Like `cast_double_to_torus_with_rounding`? We use the word `Torus` in the code base to refer to the ciphertext modulus size (u32 or u64).

Reply:
I meant the v1 and V1

Reply:
Oh indeed, could be `double_v1` and `torus_v1` for example @bbarbakadze.

Reply:
what about `double_real` `double_imag`  `torus_real` `torus_imag` v1 and v2 are still confusing

Reply:
Yep looks much better!